From f5ccd7940ee8e47cbf225f4c8db7490509b8f506 Mon Sep 17 00:00:00 2001
From: "Cao,Bing Bu" <bingbu.cao@intel.com>
Date: Thu, 9 Jul 2015 11:30:28 +0000
Subject: [PATCH 449/462] atomisp2: add unlocked_ioctl

In current atomisp driver, we did not specify the
unlocked_ioctl for v4l2_file_operations.

If no unlocked_ioctl, v4l2 will serial all the ioctls
and sometimes critical ioctl maybe blocked other ioctl.

add unlocked_ioctl to avoid this issue.

Change-Id: I88264d75fabc8f64f9788ef925c166c8a6f36874
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-38502
Signed-off-by: Cao,Bing Bu <bingbu.cao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/64099
Reviewed-by: Fang Wang <fang.c.wang@intel.com>
Reviewed-by: Jing Tao <jing.tao@intel.com>
Approver: Jing Tao <jing.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c     | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
index 4c933b2..71ec7a7 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
@@ -1285,6 +1285,7 @@ const struct v4l2_file_operations atomisp_fops = {
 	.release = atomisp_release,
 	.mmap = atomisp_mmap,
 	.ioctl = video_ioctl2,
+	.unlocked_ioctl = video_ioctl2,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl32 = atomisp_compat_ioctl32,
 #endif
@@ -1297,6 +1298,7 @@ const struct v4l2_file_operations atomisp_file_fops = {
 	.release = atomisp_release,
 	.mmap = atomisp_file_mmap,
 	.ioctl = video_ioctl2,
+	.unlocked_ioctl = video_ioctl2,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl32 = atomisp_compat_ioctl32,
 #endif
-- 
1.9.1

